application-ideas-api/src/main/java/com/xwiki/ideas/IdeasManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.ideas;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.stability.Unstable;
+
+import com.xwiki.ideas.model.VoteResult;
+
+/**
+ * @version $Id$
+ * @since 1.14
+ */
+@Role
+@Unstable
+public interface IdeasManager
+{
+    /**
+     * @param documentReference a reference to a document that contains an Idea
+     * @param pro true if the user agrees with the Idea; false otherwise
```

Comment:
The way it works right now is that if you vote "Pro" for an Idea that you already voted as such, it will be removed.

Reply:
OK, it acts like a toggle. This may be fine for the UI but it's not good for the API. You need to separate the two. I don't think developers expect to call vote again in order to remove the vote. It's not natural.